src/test/java/com/spotify/google/cloud/pubsub/client/PublisherTest.java

```diff
@@ -110,21 +117,82 @@ public void testOutstandingRequests() throws InterruptedException, ExecutionExce
 
     // Publish a message and verify that the outstanding request counter rises to 1
     final CompletableFuture<String> f1 = publisher.publish("t1", m1);
-    assertThat(publisher.outstandingRequests(), is(1));
+    assertThatWithin(5, SECONDS, publisher::outstandingRequests, is(1));
```

Comment:
Well, that bleeds more internal implementation detail into the tests, so I prefer to avoid that. The tests should be robust. I have not seen any flakiness with this approach.